modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/ScrollGestureRecognizer.java

```diff
@@ -245,15 +242,16 @@ public void notifyEndTouchEvent(long time) {
                     totalDeltaY += deltaY;
 
                     sendScrollEvent(false, centerAbsX, centerAbsY, currentTouchCount);
-                    double timePassed = ((double)time - scrollStartTime) / 1000000000;
-                    if (timePassed > 1e-4) {
+                    long nanosPassed = nanos - scrollStartNanos;
+
+                    if (nanosPassed > INITIAL_VELOCITY_THRESHOLD_NANOS) {
                         //capture radius (pytaguras) or init to variables x,y ???
                         double scrollMagnitude = Math.sqrt(deltaX * deltaX + deltaY * deltaY);
                         factorX = deltaX / scrollMagnitude;
                         factorY = deltaY / scrollMagnitude;
-                        initialInertiaScrollVelocity = scrollMagnitude / timePassed;
+                        initialInertiaScrollVelocity = scrollMagnitude / nanosPassed * NANOS_TO_SECONDS;
```

Comment:
is this correct?  shouldn't it be `scrollMagnitude / (nanosPassed * NANOS_TO_SECONDS)`?

Reply:
I think it's correct, unless you have reason to believe why it wouldn't be.  `scrollMagnitude` is a double, and the division and multiplication that follow it will be all done as doubles.

Reply:
your change is equivalent to 

(scrollMagnitude * NANOS_TO_SECONDS) / nanosPassed

is that correct?

Reply:
That would be equivalent yes.

Perhaps it is because `timePassed` was a `double` before measured in seconds while I'm now doing the calculation with nano seconds?

Reply:
Okay I see what you're getting at, your first suggestion would not be correct, it's supposed to be what you suggest in the 2nd one.  So what I want is:

     (scrollMagnitude / nanosPassed) * NANOS_TO_SECONDS

or:

     (scrollMagnitude * NANOS_TO_SECONDS) / nanosPassed

...but that's the same as what I wrote.

Reply:
But is it correct? The previous code computed a deltaTime in seconds as `((currentNanos - startNanos) / 1e9)`, and then divided the `scrollMagnitude` by that. I think that the equivalent to the old code is:

```
    scrollMagnitude / (nanosPassed * NANOS_TO_SECONDS)
```

Reply:
It turns out it is not correct, the parenthesis are indeed required, not sure how I could have missed it. I'm going to see if I can add a test to this as they're missing to ensure it does the same before/after (aside from precision).